llvm/include/llvm/ADT/APFloat.h

```diff
@@ -1483,6 +1483,19 @@ inline APFloat minimum(const APFloat &A, const APFloat &B) {
   return B < A ? B : A;
 }
 
+/// Implements IEEE 754-2019 minimumNumber semantics. Returns the smaller
+/// of 2 arguments, not propagating NaNs and treating -0 as less than +0.
+LLVM_READONLY
+inline APFloat minimumnum(const APFloat &A, const APFloat &B) {
+  if (A.isNaN())
+    return B.isNaN() ? B.makeQuiet() : B;
```

Comment:
Is there a reason to prefer B's payload to A's if both are nan?

Reply:
It is always B.

Reply:
```
return B.makeQuiet();
```

Reply:
To follow up, is it standard-conform to pick an arbitrary value when both are nan? What happens when GCC picks A?

Reply:
Yes. It's merely suggested you should try to preserve one of the payloads. You're also allowed to just lose the payload

Reply:
> ```
> return B.makeQuiet();
> ```

I am not sure that `makeQuiet` works with non-NaN number. That's why I am using ?:

Reply:
> Is there a reason to prefer B's payload to A's if both are nan?

Just as Thorsten said, I am lazy, so that there is always B in this line.
If we prefer A, we will have `if (B.isNaN())` in the first line, it is not so friendly to my brain ;)